lib/code-callers.js

```diff
@@ -27,39 +30,72 @@ Load is reported in:
 
 */
 
-let pythonCallers = []; // all pythonCallers, whether available for work or currently busy
-let availablePythonCallers = []; // FIFO queue for available pythonCallers waiting for a job/callback
-let waitingCallbacks = []; // FIFO queue for jobs/callbacks waiting for an available pythonCaller
+/** @typedef {PythonCaller | DockerCaller } CodeCaller */
+
+/**
+ * All pythonCallers, whether available for work or currently busy
+ *
+ * @type {CodeCaller[]}
+ */
+let pythonCallers = [];
+
+/**
+ * FIFO queue for available pythonCallers waiting for a job/callback
+ *
+ * @type {CodeCaller[]}
+ */
+let availablePythonCallers = [];
+
+/** @typedef {{ (err: Error | null, caller: CodeCaller): void, __load_uuid: string }} InternalCallback */
+
+/**
+ * FIFO queue for jobs/callbacks waiting for an available pythonCaller
+ * @type {Array<[string, InternalCallback]>}
+ */
+let waitingCallbacks = [];
+
+// Will be set to the appropriate class in init()
+let CodeCaller = null;
+
+let cancelWarmup = null;
 
 module.exports = {
   init() {
     debug('init()');
-    if (!config.useWorkers) {
-      return;
+    const { workersExecutionMode } = config;
```

Comment:
I renamed this file from `workers.js` to `code-callers.js` since I felt that better described what's going on here. However, I didn't rename the config options, as doing so would have made this more difficult to deploy (an unnecessary breaking change).